creusot-contracts/src/std/option.rs

```diff
@@ -560,13 +560,26 @@ extern_spec! {
             }
         }
     }
-}
 
-impl<T> Default for Option<T> {
-    #[open]
-    #[predicate]
-    fn is_default(self) -> bool {
-        pearlite! { self == None }
+    impl<T> IntoIterator for Option<T>{
+        #[ensures(self == result@)]
+        fn into_iter(self) -> IntoIter<T>;
+    }
+
+    impl<'a, T> IntoIterator for &'a Option<T>{
+        #[ensures(*self == match result@ { None => None, Some(r) => Some(*r) })]
+        fn into_iter(self) -> Iter<'a, T>;
+    }
+
+    impl<'a, T> IntoIterator for &'a mut Option<T>{
+        #[ensures(*self == match result@ { None => None, Some(r) => Some(*r) })]
+        #[ensures(^self == match result@ { None => None, Some(r) => Some(^r) })]
```

Comment:
not the current subject either but, also would be nice to either: do some "logical reborrowing" here or more generally to be able to treat these kinds of methods as "pure" and infer their specifications. For `extern_specs` we could provide the body and do inference over it instead.

Reply:
Yeah, we should provide logical reborrowing under sums...